### DISCUSSION TITLE
Sharing composables with environment variables using Layers

### DISCUSSION THREAD

**Discussion by EddyDezuraud** [nuxt/nuxt#26161] [Questions]
Hello, I'm working on a mono-repo project with the following structure:

- `/apps`: contains Nuxt apps (authentication page, administration, public pages, etc.)
- `/layers`: contains shared parts useful for all Nuxt projects

In each app, I have variables like `meta.env.DEV` or `config.public` set up in the `nuxt.config` file.

In one of my layers (`customFetch`), I'm centralizing a global configuration of custom `useFetch `to override the `baseURL `and headers (to avoid duplicating this method in each app). In this case, I need to set conditions based on variables defined in the `nuxt.config` of each project.

The code works fine once built, but in the code of my layer, the IDE considers all these variables as undefined.

How can I manage this situation to share composables containing environment variables across each project using layers?

Additionally, when using a library in a layer and an app, should it be declared in both package.json files?

Thanks in advance for your help!

**Comment by saschafuchs** [comment]
Basically, I use runtimeConfig for this purpose. In a sublayer, I utilize these variables to control conditions. In the main application, if I want to manage the sublayers, I can specify the same variables in the runtimeConfig, but with different values. These values override the previous ones and are applied globally.
